credentials/apps/badges/models.py

```diff
@@ -150,6 +150,12 @@ def save(self, *args, **kwargs):
         else:
             raise ValidationError("Cannot update BadgeRequirement for active BadgeTemplate")
 
+    def fulfill(self, username: str):
+        try:
+            progress = BadgeProgress.objects.get(template=self.template, username=username)
```

Comment:
[get_or_create](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#get-or-create)?

We don't want to drop fulfillment if there is no progress, since it is always the case for the first fulfillment.